src/app/api/core/services/policies.service.ts

```diff
@@ -0,0 +1,33 @@
+import { UserAction, UserRole } from '@api/core/types/user'
+import { Resource } from '@api/core/types/api'
+import { BaseService } from './base.service'
+import APIError from '@api/core/exceptions/api'
+import httpStatus from 'http-status'
+
+/**
+ * Resource-level policies control service for Users (IU + Client)
+ */
+export class PoliciesService extends BaseService {
```

Comment:
i dont understand what purpose this files serves. Why is it part of this PR? what functionality does it acheive?

Reply:
Using a single policy gate-keeping service like this is to make sure we do role-based access control for resources in a clean and modular way.

Reply:
Let me start by saying, I really like the approach and code itself that you've written here, in the right context this would probably get approved instantly. I feel this shouldn't be part of this PR. Here are my arguments for why:
1. this is not connected to any part of the code in these changes, so as a reviewer i have not context on this
2. the linear task has no mention of permission request system so there is no context about it why its needed an how. Instead I have to ask for clarification in code review
3. this flow is not tested as part of this PR so who knows if this is actually going to be use-case we need.

Essentially, in the context of this PR, this is just arbitrary code that was written to cover some future scenario